[avrilmohh] iP

avrilmohh opened the pull request:
**TudeeBot**

> “Either you run the day or the day runs you.” —Jim Rohn

TudeeBot makes keeping up with your tasks so much easier! It's:
- text-based
- ~~extremely~~ (did i say EXTREMELY?) intuitive
- super convenient to use

All you need to do is,
1. Download it from [here](https://github.com/avrilmohh/ip/releases/tag/A-Jar)
2. Double-*click* it
3. Start adding your tasks
4. It will start managing your task like your personal assistant 😏 

And most importantly, it is FREE to use! Isn't that wonderful??

Features:

- [x] Add 3 different tasks: todo, deadline, event
- [x] Mark and unmark tasks as complete
- [x] Saves and loads tasks from a txt file

##
If you are a Java programmer, you can use it to practice Java too. Heres the `Tudee` method, which is the main method.

```java
public class Tudee {
  public static void main(String[] args) {
          filePath = "./data/tudee.txt"
          new Tudee(filePath).run();
      }
  }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

avrilmohh committed:
renamed bot name

avrilmohh committed:
Added more chatbot functions so the user can converse with it

avrilmohh committed:
Added list function

avrilmohh committed:
Added Task.java

avrilmohh committed:
Changed logic of Tudee.java with the addition of Task.java

avrilmohh committed:
Added additional commands to the bot

avrilmohh committed:
Automated Text UI Testing

avrilmohh committed:
Added exceptions

avrilmohh committed:
Added delete function

avrilmohh committed:
Modified the code to use enumerations

avrilmohh committed:
Add new class Storage to help with saving and loading tasks

avrilmohh committed:
Move data file to correct position

avrilmohh committed:
Move data file to correct position

avrilmohh committed:
Fix formatting bugs

avrilmohh committed:
Merge branch 'branch-Level-7'

avrilmohh committed:
Modify chatbot to understand dates and times

avrilmohh committed:
Merge branch 'branch-level-8'

avrilmohh committed:
Add Ui.java

avrilmohh committed:
Change the bot to OOP

avrilmohh committed:
Merge branch 'A-MoreOOP'

avrilmohh committed:
Organise into packages

avrilmohh committed:
Merge branch 'A-Packages'

avrilmohh committed:
Merge branch 'add-gradle-support' of https://github.com/avrilmohh/ip

avrilmohh committed:
Add Gradle

avrilmohh committed:
Add test for ToDo.java and delete redundant files

avrilmohh committed:
Add JUnit tests for ToDo and Deadline

avrilmohh committed:
Merge branch 'A-JUnit'

avrilmohh committed:
Add JavaDoc

avrilmohh committed:
Modify code to adhere to coding standard

avrilmohh committed:
Add find functionality to Tudee chatbot

avrilmohh committed:
Merge tag 'A-JavaDoc'

avrilmohh committed:
Merge tag 'A-CodingStandard'

avrilmohh committed:
Merge tag 'Level-9'

XinghuaJulia commented on src/main/java/tudee/Tudee.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /**
+     * Creates the Tudee chatbot main loop
```
```suggestion
     * Creates the Tudee chatbot main loop.
```

XinghuaJulia commented on src/main/java/tudee/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+
+/**
+ * Represents a Deadline task.
+ * A Deadline task is a task with a specific deadline
```
```suggestion
 * A Deadline task is a task with a specific deadline.
```

XinghuaJulia commented on src/test/java/tudee/task/DeadlineTest.java:
```diff
@@ -0,0 +1,33 @@
+import java.time.LocalDate;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class DeadlineTest {
```
Where is the header comments? What are you testing specifically?

avrilmohh replied:
I have fixed the code comments by adding full stops, thank you for the suggestions 👍

XinghuaJulia reviewed (commented):
Good job!!! I liked how you separated the commands into their individual classes, making your code easier and more intuitive to understand. However, there are some minor design choices such as having packages with only 1 file in the package. Remember to put code comments with full stops, following the  coding standard. You can also consider using switch statements in the Parser and String.format to clean up your code more! Otherwise, good job :D

avrilmohh committed:
Fix checkstyle errors

avrilmohh committed:
Add GUI

arnav-goel05 commented on src/main/java/tudee/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @param task The task to be added.
+     */
+    public void add(Task task) {
```
Perhaps the method name could be changed `addTask` to make it more descriptive. Similarly, the method names below could also be potentially changed to represent a verb instead of a noun.

arnav-goel05 commented on src/main/java/tudee/ui/Ui.java:
```diff
@@ -0,0 +1,144 @@
+     *
+     * @param task The task that has been marked as not done.
+     */
+    public String showUnMark(Task task) {
```
Method name could be potentially changed to `showUnmarkedTask` to better describe the method.

arnav-goel05 commented on src/main/java/tudee/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+ */
+public abstract class Task {
+    protected String taskString;
+    protected boolean done;
```
Boolean variable `done` could be potentially changed to `isDone`.

arnav-goel05 commented on src/main/java/tudee/task/Events.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    @Override
+    public String toFileString() {
+        return "E | " + (done ? 1 : 0) + " | " + taskString + " | " + start + " | " + end;
```
```suggestion
        return "E | " + (done ? 1 : 0) + " | "
               + taskString + " | " + start + " | " + end;
```

arnav-goel05 commented on src/main/java/tudee/storage/Storage.java:
```diff
@@ -0,0 +1,96 @@
+        try (Scanner sc = new Scanner(currentFile)) {
+            while (sc.hasNextLine()) {
+                String currentLine = sc.nextLine();
+                String[] data = currentLine.split(" \\| ");
```
Perhaps the array `data` could be renamed to `splitString` since it is more descriptive.

arnav-goel05 reviewed (commented):
Great job on your code! Just a few styling and naming changes requested.

avrilmohh committed:
Update data

avrilmohh committed:
Merge tag 'Level-10'

avrilmohh committed:
Add assert to Parser, TaskList, Storage and Command child classes.

We did not check whether the format of the user inputs are correct.
The user may incorrectly input their commands as they are not aware of
the format which leads to unnecessary errors. To fix this, add
assertions. This checks if the user input for specific instructions
has the correct format and does not raise unnecessary errors like
IndexOutOfBoundsException.

Add try-catch for DateTimeParseException in Deadline and Events class.

It is not clear to the user which datetime format to use. There are
many different datetime formats such as MM/DD/YYYY or DD/MM/YYY.
Therefore, standardising the format to YYYY-MM-DD would make the
chatbot easier to use.

avrilmohh committed:
Improve code quality

- Some methods have unclear names. This could be confusing to the reader
  so their names have been changed for better understanding.

- Magic numbers are used in code. This is not clear to the reader on the
  purpose of these numbers. As such, named constants were used to
  replace these magic numbers.

- Some class methods have deep nesting. This makes it harder for the
  reader to keep track of the logic. To fix this, use smaller private
  methods to replace the nesting.

- There is a complicated boolean expression in DateCommand class. This
  could cause confusion so define the booleans first then test for the
  desired condition.

avrilmohh committed:
Merge pull request #1 from avrilmohh/heads/A-Assertions

Add assert and try-catch

avrilmohh committed:
Merge branch 'master' into A-CodeQuality

# Conflicts:
#	src/main/java/tudee/command/AddTaskCommand.java
#	src/main/java/tudee/command/DateCommand.java
#	src/main/java/tudee/command/DeleteCommand.java
#	src/main/java/tudee/command/FindCommand.java
#	src/main/java/tudee/command/ListCommand.java
#	src/main/java/tudee/command/MarkCommand.java
#	src/main/java/tudee/command/UnmarkCommand.java
#	src/main/java/tudee/parser/Parser.java
#	src/main/java/tudee/storage/Storage.java
#	src/main/java/tudee/task/TaskList.java

avrilmohh committed:
Merge pull request #2 from avrilmohh/heads/A-CodeQuality

Improve code quality

avrilmohh committed:
Add Reminder functionality to the chatbot. In order to receive a
reminder, the user have to input list command.

avrilmohh committed:
Merge branch 'B-Reminders'

avrilmohh committed:
Remove binary files and fix boolean name issue

avrilmohh committed:
Fix checkstyle

avrilmohh committed:
Ensure tudee can handle more common errors

avrilmohh committed:
Improve SLAP

avrilmohh committed:
Add Ui.png and fix naming

avrilmohh committed:
Update README instructions

avrilmohh committed:
Remove README comment

avrilmohh committed:
Change image directory

avrilmohh committed:
Modify DeadlineTest.java and ToDoTest.java
